main.js

```diff
@@ -68,104 +66,96 @@ var createTopLine = (post => {
-    $commentsDisplaySection.append($fullPagePost);
-    renderCommentLinesOnly(postToDisplay,$commentsDisplaySection);
-    //hide the original page and show the comments page
+
+    //hide the main page & show the comments page, then render the single post
     $fullMainPage.hide();
     $fullCommentsPage.show();
-    //clear out the comments section, hide the comments page, & show the original page
+    renderSinglePostPage(postToDisplay);
+
+    // when a user goes back, hide the post & comments page and show the main page
     $('#back-link').click(function(e) {
       e.preventDefault();
-      $commentsDisplaySection.empty();
+      //turn off the button listener because it will be re-added each time
+      $('#post-comment-button').off();
+      //clear out the comments section
+      $('#comments').empty();
       $fullMainPage.show();
       $fullCommentsPage.hide();
       renderPosts();
     })
-    $('#post-comment-button').click(function() {
+
+    //when a user posts a comment, it will render on the current page & will be stored with its post
+    $('#post-comment-button').on('click', function() {
       var newCommentText = $('#new-comment-message').val();
       var newCommentUser = $('#new-comment-name').val();
-      commentId++;
-      var newComment = {
-        commentText : newCommentText,
-        commentUser : newCommentUser,
-        commentId : commentId
-      }
-      postToDisplay.comments.push(newComment);
-      $commentsDisplaySection.empty();
-      $commentsDisplaySection.append($fullPagePost);
-      renderCommentLinesOnly(postToDisplay,$commentsDisplaySection);
+      //add the comment to the posts array, then re-render so it displays
+      addNewComment(postToDisplay, newCommentText, newCommentUser);
+      renderSinglePostPage(postToDisplay);
+      //empty the input boxes to prepare for the next comment
       $('#new-comment-message').val("");
       $('#new-comment-name').val("");
     })
-
   }
 
-  //build the line from the handlebars template
+  //build the top line for a post from the handlebars template
   var topLineSource = $('#top-line-template').html();
   var topLineTemplate = Handlebars.compile(topLineSource);
   var $topLine = $(topLineTemplate(post));
-
   //individually bind click handlers to each link
   $topLine.on('click', '.remove-post', handleRemoveClick);
   $topLine.on('click', '.comments-show-hide', toggleComments);
   $topLine.on('click', '.edit-post', handleEditClick);
-    $topLine.on('click', '.full-page-view', handleFullPageClick);
-
+  $topLine.on('click', '.full-page-view', handleFullPageClick);
   //return the line to the posting click handler to assemble with the post text
   return $topLine;
 })
 
-// get the text & user name and build a new comment into the comment array, along with pushing the comment input to the bottom
-var createCommentLine = (post => {
+// create the comment line for the main page
+var createCommentInputLine = (post => {
```

Comment:
Similar comment as I mentioned above, although, in this case, you don't even need to have this function take a parameter seeing as you aren't using it anywhere within your function.